pkg/ipamplugin/ipam_plugin.go

```diff
@@ -251,7 +255,16 @@ func cmdAdd(args *skel.CmdArgs) error {
 			assignArgs.HostReservedAttrIPv4s = rsvdAttrWindows
 		}
 		logger.WithField("assignArgs", assignArgs).Info("Auto assigning IP")
-		assignedV4, assignedV6, err := calicoClient.IPAM().AutoAssign(ctx, assignArgs)
+		autoAssignWithLock := func(calicoClient client.Interface, ctx context.Context, assignArgs ipam.AutoAssignArgs) ([]cnet.IPNet, []cnet.IPNet, error) {
```

Comment:
Think we'll also want a lock around the AssignIP call here: https://github.com/projectcalico/cni-plugin/pull/985/files#diff-2c8ea97e5c1bc5a024e2780d4b842e2b444124cf1e9670e18d91100405c7b4ddR179